interfaces/cython/cantera/reaction.pyx

```diff
@@ -1251,3 +1435,472 @@ cdef class BlowersMaselInterfaceReaction(BlowersMaselReaction):
+    instantiated from XML. It is the default for import from YAML.
+    """
+    reaction_type = "three-body"
+
+    cdef CxxThreeBodyReaction3* tbr(self):
+        return <CxxThreeBodyReaction3*>self.reaction
+
+    cdef CxxThirdBody* thirdbody(self):
+        return <CxxThirdBody*>(self.tbr().thirdBody().get())
+
+    def __init__(self, equation=None, rate=None, efficiencies=None,
+                 Kinetics kinetics=None, init=True, **kwargs):
+
+        if init and equation and kinetics:
+
+            spec = {"equation": equation, "type": self.reaction_type}
+            if isinstance(rate, dict):
+                spec["rate-constant"] = rate
+            elif isinstance(rate, ArrheniusRate) or rate is None:
+                pass
+            else:
+                raise TypeError("Invalid rate definition")
+
+            if isinstance(efficiencies, dict):
+                spec["efficiencies"] = efficiencies
+
+            self._reaction = CxxNewReaction(dict_to_anymap(spec),
+                                            deref(kinetics.kinetics))
+            self.reaction = self._reaction.get()
+
+            if isinstance(rate, ArrheniusRate):
+                self.rate = rate
+
+    property efficiencies:
+        """
+        Get/Set a `dict` defining non-default third-body efficiencies for this
+        reaction, where the keys are the species names and the values are the
+        efficiencies.
+        """
+        def __get__(self):
+            return comp_map_to_dict(self.thirdbody().efficiencies)
+        def __set__(self, eff):
+            self.thirdbody().efficiencies = comp_map(eff)
+
+    property default_efficiency:
+        """
+        Get/Set the default third-body efficiency for this reaction, used for
+        species used for species not in `efficiencies`.
+        """
+        def __get__(self):
+            return self.thirdbody().default_efficiency
+        def __set__(self, default_eff):
+            self.thirdbody().default_efficiency = default_eff
+
+    def efficiency(self, species):
+        """
+        Get the efficiency of the third body named *species* considering both
+        the default efficiency and species-specific efficiencies.
+        """
+        return self.thirdbody().efficiency(stringify(species))
+
+
+cdef class PlogReaction3(Reaction3):
+    """
+    A pressure-dependent reaction parameterized by logarithmically interpolating
+    between Arrhenius rate expressions at various pressures.
+
+    An example for the definition of an `PlogReaction3` object is given as::
+
+        rxn = PlogReaction3(equation='H2 + O2 <=> 2 OH', rate={"rate-constants":
+            [{'P': 1013.25, 'A': 1.2124e+16, 'b': -0.5779, 'Ea': 45491376.8},
+             {'P': 101325., 'A': 4.9108e+31, 'b': -4.8507, 'Ea': 103649395.2},
+             {'P': 1013250., 'A': 1.2866e+47, 'b': -9.0246, 'Ea': 166508556.0},
+             {'P': 10132500., 'A': 5.9632e+56, 'b': -11.529, 'Ea': 220076726.4}]})
```

Comment:
This is by design. My objective here was that all `*Reaction` constructors use the same keywords: `equation`, `rate` and `kinetics` (the last was erroneously missing here, which is fixed). For `rate`, the accepted value would be either a `*Rate` object, or a dictionary that holds arguments for the `*Rate` constructor.

Reply:
I guess my thinking on this was that the most useful form of consistency would be to have these constructors parallel the YAML format, where the parameters for the rate constant are at the same level as the `equation`, rather than introducing a different way of arranging the same data that's more tied to the specific implementation.

That's essentially what you have for the elementary reaction -- you don't make the user pass the argument as `rate={"rate-constant": {"A": ..., "b": ..., "Ea": ...}}`. In that case, the only difference is in the name `rate` as opposed to `rate-constant`, where something had to change anyway given the `-` in the identifier.

I do like the idea of being able to pass in an already-instantiated `ReactionRate` object, but I don't think that these two ideas are mutually exclusive.

Reply:
Good catch - this definitely wasn't consistent. When I implemented this, I wanted to pass the argument list for the constructor of the corresponding `*Rate` object as the `rate` argument, but apparently never finished. I now fixed this, with the input:
```Python
        rxn = PlogReaction(
            equation="H2 + O2 <=> 2 OH",
            rate=[(1013.25, Arrhenius(1.2124e+16, -0.5779, 45491376.8)),
                  (101325., Arrhenius(4.9108e+31, -4.8507, 103649395.2)),
                  (1013250., Arrhenius(1.2866e+47, -9.0246, 166508556.0)),
                  (10132500., Arrhenius(5.9632e+56, -11.529, 220076726.4))],
            kinetics=gas)
``` 
For the YAML equivalent, I updated `Reaction.fromYaml` so it can be created from arguments that correspond to YAML fields. I.e. both variants are now functional.